### DISCUSSION TITLE
error compiling STAR2 on Mac with an M1 chip

### DISCUSSION THREAD

**Discussion by vleifer685** [alexdobin/STAR#1739] [General]
Hi all,

Running into the following problem compiling STAR2 on a Mac with an M1 chip, which is sadly incompatible with many programs. I am wondering if STAR2 is also incompatible with the M1 chip, or if someone could help me troubleshoot.

I run into the following issue compiling after running "make STAR" in the "source" folder within STAR-2.7.10b. I have brew fully upgraded so I don't think this is the problem.

date: illegal option -- -
usage: date [-jnRu] [-r seconds|file] [-v[+|-]X[ymwdHMS]]
            [-I[date | hours | minutes | seconds]]
            [-f fmt date | [[[mm]dd]HH]MM[[cc]yy][.ss]] [+format]
c++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=" :/Users/X/STAR-2.7.10b/source"' -D'GIT_BRANCH_COMMIT_DIFF=""' -pipe -Wall -Wextra systemFunctions.cpp
clang: error: unsupported option '-fopenmp'
make: *** [systemFunctions.o] Error 1

**Comment by alexdobin** [comment]
Hi @vleifer685 

to compile STAR on any Mac, you need to specify the path to brew's g++, e.g.
```make STARforMacStatic CXX=/usr/local/Cellar/gcc/8.2.0/bin/g++-8```

However, I do not have experience compiling on M1 Macs, and it may not work even with g++.
Not sure if anyone has done it successfully.

**Comment by l-yampolsky** [comment]
Similar issue: can't compile on Mac with M1. At first, I had the same problem, clang: error: unsupported option '-fopenmp', forgetting to specify path to g++. Corrected that.
Then, upon
make STARforMacStatic CXX=/opt/homebrew/Cellar/gcc/12.2.0/bin/g++-12 
getting:
[...]
g++-12: error: unrecognized command-line option '-mavx2'
make: *** [opal/opal.o] Error 1

Any idea by any chance what is it?
Thanks!
Would be great to be able to get STAR to work on new Macs!

**Comment by kayihui** [comment]
Hi there,

I am able to use pre-compiled STAR (v2.7.9a) on the new Mac M1 Max.
 
You can switch the terminal from ARM to x86_64 mode.

Check the terminal whether it's running with ARM or x86_64:
uname -m

If the output is arm64, your terminal is running on an Apple M1 (ARM) Mac. If the output is x86_64, your terminal is running on an Intel-based Mac.

Convert to x86_64: 
arch -x86_64 /bin/zsh

Hope this help!

**Comment by l-yampolsky** [comment]
This helps a lot, thanks!

2.7.10b seems to be running as well.


Best,


L.Yampolsky


________________________________
From: kayihui ***@***.***>
Sent: Thursday, March 2, 2023 6:17 AM
To: alexdobin/STAR ***@***.***>
Cc: Yampolsky, Lev ***@***.***>; Comment ***@***.***>
Subject: [EXTERNAL] Re: [alexdobin/STAR] error compiling STAR2 on Mac with an M1 chip (Discussion #1739)


Hi there,

I am able to use pre-compiled STAR (v2.7.9a) on the new Mac M1 Max.

You can switch the terminal from ARM to x86_64 mode.

Check the terminal whether it's running with ARM or x86_64:
uname -m

If the output is arm64, your terminal is running on an Apple M1 (ARM) Mac. If the output is x86_64, your terminal is running on an Intel-based Mac.

Convert to x86_64:
arch -x86_64 /bin/zsh

Hope this help!

—
Reply to this email directly, view it on GitHub<https://github.com/alexdobin/STAR/discussions/1739#discussioncomment-5177754>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/ACDXHNPZR5PRCC3GMORNWO3W2B6UDANCNFSM6AAAAAAT3KNMXU>.
You are receiving this because you commented.Message ID: ***@***.***>

The [EXTERNAL] tag in the subject line identifies emails that do NOT originate from an ETSU person or service. Please exercise caution when handling emails from external sources. Any email that is unsolicited and requires you to take immediate action, appears to be forged or is PHISHING for information can be verified by emailing the ITS Help Desk.

**Comment by kayihui** [comment]
I am glad that it helped!

I am curious how's the performance from your side.

It seems to me running on the M1 Max 32G Ram is slower than 8-Core Intel i9 with 64G Ram ( ~100M/hr Vs ~600M/hr in the log).

@alexdobin,  I wonder if the amount of Ram is the main speed contributor? Or I need to put in higher number of thread?

I have the impression that the M1 Max is not "working hard enough" (fan is not on and the speed slow down when the screen go off), while the intel i9, fan was up real quick and the mapping is done way faster.

**Comment by alexdobin** [comment]
Hi @kayihui 

32GB of RAM could be a factor if the genome index size is close to 32GB.
Another explanation: the executable was compiled on x64 Mac, and needs to run through Rosetta on M1 Mac.
